library/core/src/arch.rs

```diff
@@ -17,6 +20,20 @@ pub macro asm("assembly template", $(operands,)* $(options($(option),*))?) {
     /* compiler built-in */
 }
 
+/// Inline assembly used in combination with `#[naked]` functions.
+///
+/// Refer to [Rust By Example] for a usage guide and the [reference] for
+/// detailed information about the syntax and available options.
+///
+/// [Rust By Example]: https://doc.rust-lang.org/nightly/rust-by-example/unsafe/asm.html
+/// [reference]: https://doc.rust-lang.org/nightly/reference/inline-assembly.html
+#[unstable(feature = "naked_functions", issue = "90957")]
+#[macro_export]
+#[cfg(bootstrap)]
+macro_rules! naked_asm {
+    ($(arguments,)*) => { core::arch::asm!($(arguments,)* options(noreturn)) }
```

Comment:
Try something like

```suggestion
    ($(arguments:tt)*) => { core::arch::asm!($(arguments)*, options(noreturn)) }
```

This will forward all tokens from the input and add `, options(noreturn)` as suffix.